pypeit/pypeit_steps.py

```diff
@@ -0,0 +1,914 @@
+    caliBrate = load_calibrations_for_frame(
+        spectrograph, fitstbl, par, frames[0], det, calib_ID, calibrations_path)
+
+    msgs.info(f'Reducing detector {det}')
+
+    # Instantiate Reduce object
+    # Required for pypeline specific object
+    # At instantiaton, the fullmask in self.sciImg is modified
+    objFind = instantiate_objfind(sciImg, spectrograph, fitstbl,
+                                  par, frames, det, caliBrate,
+                                  bkg_redux,
+                                  find_negative, show=show)
+    # Do it
+    initial_sky, sobjs_obj = objFind.run(std_trace=std_trace, 
+                                         show_peaks=show)
+
+    return initial_sky, sobjs_obj, objFind
+
+
+def load_calibrations_for_frame(spectrograph, fitstbl, par, frame, det, 
+                                calib_ID, calibrations_path:str):
+    """
+    Load calibrations for a specific frame and detector.
+
+    This function initializes the Calibrations class, runs the calibration steps,
+    and ensures that the calibrations are successfully loaded for the given frame
+    and detector.
+
+    Args:
+        spectrograph (:class:`~pypeit.spectrographs.spectrograph.Spectrograph`):
+            The spectrograph instance
+        fitstbl (:class:`~pypeit.metadata.PypeItMetaData`):
+            The class holding the metadata for all the frames in this PypeIt run.
+        par (:class:`~pypeit.par.pypeitpar.PypeItPar`):
+            The parameter set for the reduction, including slitmask and
+            object finding parameters.
+        frame (:obj:`int`):
+            The index of the frame in the fitstbl
+        det (:obj:`int`):
+            Detector number (1-indexed)
+        calib_ID (:obj:`str`):
+            The calibration group ID
+        calibrations_path (:obj:`str`):
+            Path to the calibration files
+
+    Returns:
+        :class:`~pypeit.calibrations.Calibrations`:
+            The loaded calibrations for the specified frame and detector.
+
+    Raises:
+        PypeItError: If the calibrations for the specified detector are unsuccessful.
+    """
+
+    # Instantiate Calibrations class
+    user_slits = slittrace.merge_user_slit(par['rdx']['slitspatnum'],
+                                            par['rdx']['maskIDs'])
+    caliBrate = calibrations.Calibrations.get_instance(
+        fitstbl, par['calibrations'], spectrograph,
+        calibrations_path, 
+        calib_ID, frame, det,
+        reuse_calibs=True, user_slits=user_slits,
+        chk_version=par['rdx']['chk_version'])
+    #caliBrate.set_config(frame, det, par['calibrations'])
+    caliBrate.run_the_steps(reload_only=True)
+
+    if not caliBrate.success:
+        msgs.error(f'Calibrations for detector {det} were unsuccessful!  The step '
+                    f'that failed was {caliBrate.failed_step}.')  
+
+    return caliBrate
+
+
+def load_skyregions(initial_slits=False, scifile=None, frame=None):
```

Comment:
This method needs some adjustment. As you copied this from `pypeit.py` there are now some variables that do not exist, like `self.par`, `self.fitstbl`, `self.objtype`.

Reply:
gosh, that was a mess

fixed (hopefully)